internal/adminapi/client.go

```diff
@@ -28,6 +29,7 @@ type Client struct {
 	lastConfigSHA       []byte
 	lastCacheStoresHash store.SnapshotHash
 
+	lock sync.RWMutex
```

Comment:
Can we name this specifically to mention that its responsible for the `lastConfigSHA`?

Also these types of locks are typically defined above the guarded field, hence the suggestion

```suggestion
	locklastConfigSHA   sync.RWMutex
	lastConfigSHA       []byte

	lastCacheStoresHash store.SnapshotHash
```